public/electron.js

```diff
@@ -602,6 +602,89 @@ ipcMain.handle('shutdown-discord-rpc', () => {
   discordRPC.shutdownRPC();
 });
 
+ipcMain.handle('download-optedout-mods', async (e, { mods, modDestFile }) => {
+  let win = new BrowserWindow();
+
+  const mainWindowListener = () => {
+    if (win) {
+      win.removeAllListeners();
+      win.close();
+      win = null;
+    }
+  };
+
+  win.once('close', () => {
+    win = null;
+  });
+
+  let cleanupFn = null;
+  win.once('closed', () => {
+    mainWindow.webContents.send('opted-out-window-closed-unexpected');
+    if (cleanupFn) cleanupFn(`Window has been closed unexpectedly`);
+    mainWindow.removeListener('closed', mainWindowListener);
+  });
+
+  mainWindow.on('closed', mainWindowListener);
+
+  for (const mod of mods) {
+    if (!win) return;
+    const { modManifest, addon } = mod;
+    try {
+      // eslint-disable-next-line no-loop-func
+      await new Promise((resolve, reject) => {
+        const modUrlDownloadPage = `https://www.curseforge.com/minecraft/mc-mods/${addon.slug}/download`;
```

Comment:
Is this a lazy attempt to scrape download urls from the /download project page? If so, you're going to have issues about MC versions not being respected, especially if you're trying to, say, download a 1.16 mod that has a 1.18 version.

Reply:
Already ran into this issue, when trying to install one of my own modpacks FancyMenu downloads the latest 1.18 version as appose to the 1.16 version :(

Reply:
ok need to find a fix for this too

Reply:
also this will fail when a modpack has an opted out resourcepack, since the url is different.
Probably it's better to get the websiteurl and adding /download/(fileid) [like polymc does](https://github.com/PolyMC/PolyMC/blob/147bde5294626d7ba1a6cd672200f7da878eddd9/launcher/modplatform/flame/FileResolvingTask.cpp#L117)

Reply:
It looks like the project page `/files/all` lets you scrape download links for any of the files in the project, and there's some sort of `filter-game-version` variable you can use to look for MC versions, although how they're currently filtering MC versions is not immediately obvious to me.
Here's an example link that requests all versions of **Not Enough Items** for **MC 1.17.1 only**: https://www.curseforge.com/minecraft/mc-mods/jei/files/all?filter-game-version=2020709689%3A8516

Reply:
could you tell me which modpack with opted-out resource packs did you try?

Reply:
https://github.com/gorilla-devs/GDLauncher/pull/1345/commits/295e9ac12900faff9618c29fbca53517b95e772d it's simpler than this, get the downloadurl from the API and add /download/(fileid) to it

Reply:
you mean the `websiteUrl`

Reply:
So i just tested again with https://github.com/Ladvace/GDLauncher/commit/295e9ac12900faff9618c29fbca53517b95e772d and the mods downloaded were 1:1 for the modpack, 

One thing I did come across which completely froze up the system was if a mod in a modpack gets removed off CF the opt'd out mod list just freezes, maybe adding a check for a 404 not found page or a manual "skip mod" button should it happen, it's probably an extremely rare case scenario, but it just hit me twice with Macaws Bridges addon mods for environmental and buzzier bees as an example.